docs/content/docs/ops/state/checkpoint_vs_savepoint.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Checkpoint VS Savepoint"
+weight: 10
+type: docs
+aliases:
+  - /ops/state/checkpoint_vs_savepoint.html
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Checkpoint VS Savepoint
```

Comment:
```suggestion
# Checkpoints vs. savepoints

## Overview

[Savepoints]({{< ref "docs/ops/state/savepoints" >}}) and [checkpoints]({{< ref "docs/ops/state/checkpoints" >}}) are two
different ways to snapshot an application's state. The differences arise as both address different use cases which 
come with different design goals and therefore different optimizations.

The primary use case for checkpoints is to provide a recovery mechanism in case of unexpected job failures. 
The [checkpoint lifecycle]({{< ref "docs/dev/datastream/fault-tolerance/checkpointing" >}}) is configurable but 
managed by Flink, i.e., checkpoints are created, owned, and released by Flink - without user interaction. More 
specifically, checkpoints are created periodically to allow fast state recovery at any time and are eventually deleted 
when no longer needed. The main design goal is to make checkpoints as fast as possible to be created and restored.
The optimizations, however, limit their flexibility such that applications must not change between creation and recovery.

Checkpoints have the following characteristics:
- fast to create and restore for fast failure recovery
- limited flexibility/usability for other use cases
- managed by Flink runtime
- stored in state backend-specific (native) data format (may be incremental depending on the specific backend)

Note: Checkpoints are automatically deleted if the application is terminated by the user (except if checkpoints are 
explicitly configured to be retained).

Although [savepoints]({{< ref "docs/ops/state/savepoints" >}}) are created internally with the same mechanisms as 
checkpoints, they are conceptually different and designed for flexibility and portability in application maintenance 
and operation. The primary use case is a planned and manual backup and resume of an application's state.
Savepoints enable, e.g., Flink version updates, job graph changes, configuration changes like parallelism, or forking a second job as in red/blue deployment.
As a consequence, savepoints are slower to create and restore than checkpoints. Also savepoints are created, owned 
and deleted solely by the user. That means, Flink does not delete savepoints neither after job termination nor after 
restore.

Savepoints have the following characteristics:
- slower to create and restore than checkpoints
- support flexibility and portability for application maintenance and operation
- managed by the user
- stored in a state backend independent (canonical) format (Note: Since Flink 1.15, savepoints can be also stored in 
the backend-specific [native]({{< ref "docs/ops/state/savepoints" >}}#savepoint-format) format which is faster to create 
and restore but comes with some limitations.) 

## Capabilities and limitations

The following table gives an overview of capabilities and limitations for the various types of savepoints and 
checkpoints.
```
```